File: app/src/main/java/de/openfiresource/falarm/models/api/OpenPagerService.java
package de.openfiresource.falarm.models.api;

import io.reactivex.Completable;
import io.reactivex.Single;
import okhttp3.ResponseBody;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface OpenPagerService {

    @POST("auth/login/")
    Single<UserKey> login(@Body UserLogin user);

    @POST("auth/logout/")
    Completable logout();

    @GET("auth/user/")
    Single<User> getUser();

    @PUT("auth/user/")
    Single<User> putUser();

    @PUT("devices/{token}/")
    Completable putDeviceInfo(@Path("token") String token, @Body Device deviceInfo);
}

File: app/src/main/java/de/openfiresource/openpager/models/api/OpenPagerService.kt
package de.openfiresource.openpager.models.api

import io.reactivex.Completable
import io.reactivex.Single
import retrofit2.http.*

interface OpenPagerService {

    @get:GET("auth/user/")
    val user: Single<User>

    @POST("auth/login/")
    fun login(@Body user: UserLogin): Single<UserKey>

    @POST("auth/logout/")
    fun logout(): Completable

    @PUT("auth/user/")
    fun putUser(): Single<User>

    @PUT("v1/devices/{token}/")
    fun putDeviceInfo(@Path("token") token: String, @Body deviceInfo: Device): Completable
}
